app/models/miq_worker.rb

```diff
@@ -318,16 +316,45 @@ def send_message_to_worker_monitor(message, *args)
     )
   end
 
+  def self.before_fork
+    preload_for_worker_role if respond_to?(:preload_for_worker_role)
+  end
+
+  # When we fork, the children inherits the parent's file descriptors
+  # so we need to close any inherited raw pg sockets in the child.
+  def self.close_pg_sockets_inherited_from_parent
+    owner_to_pool = ActiveRecord::Base.connection_handler.instance_variable_get(:@owner_to_pool)
+    owner_to_pool[Process.ppid].values.compact.each do |pool|
+      pool.connections.each do |conn|
+        socket = conn.raw_connection.socket
+        _log.info "Closing socket: #{socket}"
+        IO.for_fd(socket).close
```

Comment:
After we fork, let's close the parent's inherited pg socket file descriptors in the child